### PR TITLE
[docs] Add information about L1/L2 addresses

### PR DESCRIPTION
**Pull request by amayvs**
- Reduce ambiguity in usage of L1/L2 addresses
- Explain how keys are generated and used

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -61,3 +61,27 @@ groovy order.groovy bench
 cd typescript
 yarn run bench
 ```
+
+## Ethereum and Starknet Keys
+
+Ethereum (L1) private key is used to sign a signature. This signature is deterministic
```
Comment (luccamordente) [inline README.md:67]:
```suggestion
Ethereum (L1) private key is used to sign a message. This signature is deterministic
```

File: README.md [hunk 2]
```diff
@@ -61,3 +61,27 @@ groovy order.groovy bench
...
+and can only be generated by the owner of the private key.
+
+This signature is then used to generate Starknet (L2) private key
+(also referred to as Paradex private key).
+
+Starknet (L2) private key can used to deduce Starknet public key.
+This private key is also used to sign auth, onboarding and order signatures.
```
Comment (luccamordente) [inline README.md:74]:
```suggestion
This private key is also used to sign auth, onboarding and order messages.
```
Another option is to say "used to generate auth, onboarding and order signatures"

File: README.md [hunk 3]
```diff
@@ -61,3 +61,27 @@ groovy order.groovy bench
...
+
+Auth endpoint also requires providing the Ethereum address which is derived
+from the Ethereum private key in examples of some languages.
+
+During the onboarding process, an Account Contract is deployed on Starknet
+with the Starknet public key. This contract is used to store the user's balances
+and this is what you see in UI as your wallet.
```
Comment (luccamordente) [inline README.md:81]: > and this is what you see in UI as your wallet

What is `this` referring to? It seems that it's referring to the public key. The public key however is not available on the UI. What is available is the contract address. If not referring to the public key, could you clarify?

**Review by luccamordente — APPROVED** [review]
Great clarification. Thanks! 
Leaving a few questions and suggestions.

**Review by luccamordente — APPROVED** [review]
LGTM! 🚢

File: README.md [hunk 4]
```diff
@@ -61,3 +61,27 @@ groovy order.groovy bench
...
+## Ethereum and Starknet Keys
+
+Ethereum (L1) private key is used to sign a message. Generated signature is
+deterministic and can only be generated by the owner of the Ethereum private key.
+
+This signature is then used to generate Starknet (L2) private key
+(also referred to as Paradex private key).
+
+Starknet (L2) private key can used to deduce Starknet public key.
+L2 private key is also used to generate auth, onboarding and order signatures.
```
Comment (sv) [inline README.md:74]: is used to sign auth, onboarding and order requests

File: README.md [hunk 5]
```diff
@@ -61,3 +61,27 @@ groovy order.groovy bench
...
+
+Auth endpoint also requires providing the Ethereum address which is derived
```
Comment (sv) [inline README.md:76]: Remove `also`. What does this mean - "in examples of some languages"?

File: README.md [hunk 6]
```diff
@@ -61,3 +61,27 @@ groovy order.groovy bench
...
+from the Ethereum private key in examples of some languages.
+
+During the onboarding process, an Account Contract is deployed on Starknet
+with the Starknet public key. This account contract address is used to identify
+token and perpetual balances on chain.
+
+On UI, after connecting with your Ethereum wallet, you will be able to click on
```
Comment (sv) [inline README.md:83]: UI -> Paradex Webapp

**Pull request merged**